src/pages/MovieDetails.vue

```diff
@@ -15,44 +36,211 @@ export default defineComponent({
   },
   data() {
     return {
-      movieDetails: null,
-      isLoading: true,
-      errorMessage: null,
+      movie: null,
+      movieIsLoading: true,
+      movieError: null,
+      date: new Date(),
     };
   },
   computed: {
+    ...mapState(useSeancesStore, [
+      "seances",
+      "seancesIsLoading",
+      "seancesError",
+      "getSeancesErrorMessage",
+    ]),
+    getMovieErrorMessage() {
+      return (
+        this.movieError?.message ||
+        "We are sorry, but the movie cannot be displayed"
+      );
+    },
     getMovieTitle() {
-      return this.movieDetails.title;
+      return this.movie.title;
+    },
+    getMovieReleaseYear() {
+      return this.movie.release_date.substring(0, 4);
+    },
+    getImagePath() {
+      return this.movie.poster_url;
+    },
+    getStyledImage() {
+      return { background: "url(" + this.getImagePath + ") center / cover" };
+    },
+    getFormattedWeekdayAndDate() {
+      const weekday = this.date.toLocaleDateString("en-US", {
+        weekday: "long",
+      });
+      const formattedDate = this.date.toLocaleDateString("en-GB");
+      return `${weekday} ${formattedDate}`;
+    },
+    formattedDate() {
+      return this.date.toISOString().substring(0, 10);
+    },
+  },
+  watch: {
+    date(newDate, oldDate) {
+      if (newDate !== oldDate) {
+        this.getSeances(this.formattedDate);
+      }
     },
   },
   methods: {
-    async getMovieDetails() {
-      this.isLoading = true;
+    async getMovie() {
+      this.movieIsLoading = true;
       try {
         const respData = await getMovieById(this.movieId);
-        this.movieDetails = respData.data;
+        this.movie = respData.data;
       } catch (error) {
-        this.errorMessage = error.message;
+        this.movieError = error;
       } finally {
-        this.isLoading = false;
+        this.movieIsLoading = false;
       }
     },
+    ...mapActions(useSeancesStore, ["getSeances"]),
+    movieSeances(movieId) {
+      return this.seances.filter((seance) => seance.movie === movieId);
+    },
   },
   mounted() {
-    this.getMovieDetails();
+    this.getMovie();
+    this.getSeances(this.formattedDate);
   },
 });
 </script>
 
 <template>
-  <div>
-    <h1>Movie details page</h1>
-    <LoadingData v-if="isLoading" />
-    <ErrorMessage v-else-if="errorMessage">{{ error.message }}</ErrorMessage>
-    <div v-else class="movie__details">
-      <div>{{ getMovieTitle }}</div>
-    </div>
-  </div>
+  <LoadingData v-if="movieIsLoading" />
```

Comment:
You can also span conditional elements with `<template v-if=“”>` if you have more elements depending on a condition. Not that it is needed here. Just a tip. 💡